app/src/main/java/com/locoquest/app/MainActivity.kt

```diff
@@ -101,25 +107,19 @@ class MainActivity : AppCompatActivity(), OnMapReadyCallback {
                 try {
                     val credential = oneTapClient.getSignInCredentialFromIntent(data)
                     val idToken = credential.googleIdToken
-                    if (idToken != null) {
-                        // Got an ID token from Google. Use it to authenticate
-                        // with Firebase.
+                    idToken?.let{
+                        // Got an ID token from Google. Use it to authenticate with Firebase.
                         val firebaseCredential = GoogleAuthProvider.getCredential(idToken, null)
                         auth.signInWithCredential(firebaseCredential)
                             .addOnCompleteListener(this) { task ->
                                 if (task.isSuccessful) {
                                     // Sign in success, update UI with the signed-in user's information
-                                    Log.d(TAG, "signInWithCredential:success")
-                                    //TODO: hideSignInButton()
-                                    Log.d(TAG, "Got ID token.")
+                                    Log.d(TAG, "You are logged in")
                                 } else {
                                     // If sign in fails, display a message to the user.
-                                    Log.w(TAG, "signInWithCredential:failure", task.exception)
-                                }
+                                    Log.w(TAG, "Error logging in" + task.exception)
```

Comment:
Messages for users should be more simple and more understandable. (https://kotlinlang.org/docs/exceptions.html#java-interoperability)

Reply:
Logging won't be seen by a user, though.  It will typically be seen by support personnel or software developers, who are technically savvy.  In that  case, it makes sense to have something that is easy to search for across logs.

Also, careful when removing TODOs.  The IDE will warn you of any incomplete TODOs when you commit and push, so it's a good reminder of work to be done.  Removing these may lead developers to neglect and/or forget about the work.